### DISCUSSION TITLE
ArgoCD UI | How can we cache the main.js file?

### DISCUSSION THREAD

**Discussion by EugeneLugovtsov** [argoproj/argo-cd#8585] [Q&A] [answered]
![изображение](https://user-images.githubusercontent.com/34510252/155000358-2a5703bd-83f8-4477-a5b0-2593b38ef975.png)

Is it possible to somehow cache this file so that it does not load every time?

**Comment by reiniertimmer** [comment, marked as answer]
I think it can be fixed by adding an annotation to the ingress rule, so that a cache header gets added for this file by the ingress controller.

Like this https://stackoverflow.com/questions/52711061/setting-cache-control-headers-for-nginx-ingress-controller-on-kubernetes-gke

I did some initial testing and it seems to work quite well. I will have to make sure that this doesn't have any unwanted side effects though

**Reply by EugeneLugovtsov** [reply]
Yeap, thanks for the tip, I added the following annotations and it works like a charm
```
nginx.ingress.kubernetes.io/configuration-snippet : |
        if ($request_uri ~* main\.[a-z0-9]*\.js) {
          expires 1M;
          add_header Cache-Control "public";
          add_header Cache-Control "immutable";
        }
```
